bson-kotlinx/src/main/kotlin/org/bson/codecs/kotlinx/DateTimeSerializers.kt

```diff
@@ -0,0 +1,224 @@
+/*
+ * Copyright 2008-present MongoDB, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.bson.codecs.kotlinx
+
+import java.time.ZoneOffset
+import kotlinx.datetime.Instant
+import kotlinx.datetime.LocalDate
+import kotlinx.datetime.LocalDateTime
+import kotlinx.datetime.LocalTime
+import kotlinx.datetime.TimeZone
+import kotlinx.datetime.UtcOffset
+import kotlinx.datetime.atDate
+import kotlinx.datetime.atStartOfDayIn
+import kotlinx.datetime.toInstant
+import kotlinx.datetime.toLocalDateTime
+import kotlinx.serialization.ExperimentalSerializationApi
+import kotlinx.serialization.KSerializer
+import kotlinx.serialization.SerializationException
+import kotlinx.serialization.descriptors.PrimitiveKind
+import kotlinx.serialization.descriptors.PrimitiveSerialDescriptor
+import kotlinx.serialization.descriptors.SerialDescriptor
+import kotlinx.serialization.encoding.Decoder
+import kotlinx.serialization.encoding.Encoder
+import kotlinx.serialization.modules.SerializersModule
+import kotlinx.serialization.modules.plus
+import org.bson.BsonDateTime
+import org.bson.codecs.kotlinx.utils.SerializationModuleUtils.isClassAvailable
+
+/**
+ * The default serializers module
+ *
+ * Handles:
+ * - ObjectId serialization
+ * - BsonValue serialization
+ * - Instant serialization
+ * - LocalDate serialization
+ * - LocalDateTime serialization
+ * - LocalTime serialization
+ */
+@ExperimentalSerializationApi
+public val dateTimeSerializersModule: SerializersModule by lazy {
+    var module = SerializersModule {}
+    if (isClassAvailable("kotlinx.datetime.Instant")) {
+        module += InstantAsBsonDateTime.serializersModule
+    }
+    if (isClassAvailable("kotlinx.datetime.LocalDate")) {
+        module += LocalDateAsBsonDateTime.serializersModule
+    }
+    if (isClassAvailable("kotlinx.datetime.LocalDateTime")) {
+        module += LocalDateTimeAsBsonDateTime.serializersModule
+    }
+    if (isClassAvailable("kotlinx.datetime.LocalTime")) {
+        module += LocalTimeAsBsonDateTime.serializersModule
+    }
+    module
+}
+
+/**
+ * Instant KSerializer.
+ *
+ * <p>
+ * Encodes and decodes {@code Instant} objects to and from {@code BsonDateTime}. Data is extracted via {@link
+ * Instant#fromEpochMilliseconds()} and stored to millisecond accuracy. </p>
+ *
+ * @since 5.2
+ */
```

Comment:
Note Kotlin uses dokka and [kdoc syntax](https://kotlinlang.org/docs/kotlin-doc.html) not JavaDoc.

```suggestion
/**
 * Instant KSerializer.
 *
 * Encodes and decodes `Instant` objects to and from `BsonDateTime`. Data is extracted via
 * [kotlinx.datetime.Instant.fromEpochMilliseconds] and stored to millisecond accuracy.
 *
 * @since 5.2
 */
```